### PR TITLE
(dbt) Migrate enriched_offer_data

### PR DESCRIPTION
**Pull request by carolinemestre-passculture**
# DE PR

## Describe your changes

The purpose of this PR is to migrate the `enriched_offer_data` model, the new name of the model will be `mrt_global__offer` and both tables must be identical.

We are also migrating/updating all the necessary models for the construction of this model.

Tag a reviewer if necessacy  @LucileRainteau 

## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [ ] New table (non-breaking change)
- [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
      
### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] My code passes CI/CD tests
- [x] I updated README.md
- [ ] I have updated the dag
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?
- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] ⏰ no, but I created a ticket

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__venue.sql [hunk 1]
```diff
@@ -61,7 +61,6 @@ SELECT
     v.venue_siret,
     v.venue_latitude,
     v.venue_longitude,
-    v.venue_managing_offerer_id,
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__venue.sql:64]: Ici on a l'offerer_id qu'on récupère aussi via le inner join avec le modèle `int_applicative__offerer`.

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__stock.sql [hunk 2]
```diff
@@ -2,15 +2,18 @@ WITH bookings_grouped_by_stock AS (
 
     SELECT
         stock_id,
+        -- garder que un des deux entre total_bookings et total_individual_bookings
         SUM(CASE WHEN booking_is_cancelled = False THEN booking_quantity END) AS total_bookings,
         COUNT(booking_id) AS total_individual_bookings,
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__stock.sql:7]: Garder que un seul des deux pour n'avoir qu'une seule façon de calculer `total_individual_bookings` cc @MatthieuRipollPassCulture

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql [hunk 3]
```diff
@@ -1,36 +1,59 @@
+{% set target_name = target.name %}
+{% set target_schema = generate_schema_name("analytics_" ~ target_name) %}
+
+{{ config(
+    pre_hook="{{create_humanize_id_function()}}"
+) }}
+
 WITH stocks_grouped_by_offers AS (
         SELECT offer_id,
             SUM(available_stock) AS available_stock,
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql:10]:
```suggestion
            SUM(available_stock) AS total_available_stock ,
```

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql [hunk 4]
```diff
@@ -1,36 +1,59 @@
...
+    GROUP BY offerId
 )
 
 SELECT
     o.offer_id,
     o.offer_id_at_providers,
+    (o.offer_id_at_providers IS NOT NULL) AS is_synchronised,
     o.offer_modified_at_last_provider_date,
     DATE(o.offer_creation_date) AS offer_creation_date,
     o.offer_creation_date AS offer_created_at,
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql:42]: redondant avec le champs au-dessus

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql [hunk 5]
```diff
@@ -45,29 +68,85 @@ SELECT
     o.offer_subcategoryId,
     o.offer_withdrawal_delay,
     o.booking_contact,
+    o.author,
+    o.performer,
+    o.stageDirector,
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql:73]:
```suggestion
    o.stageDirector as stage_director,
```

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__mediation.sql [hunk 6]
```diff
@@ -10,5 +17,7 @@ SELECT
     credit,
     isActive,
     fieldsUpdated,
```
Comment (LucileRainteau) [inline orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__mediation.sql:19]:
```suggestion
    isActive as is_active,
    fieldsUpdated as fields_updated,
```

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql (hunk 3 above)
Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql:10]: Yes bien vu !

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql [hunk 7]
```diff
@@ -45,29 +68,85 @@ SELECT
...
+    o.theater_movie_id,
+    o.theater_room_id,
+    o.speaker,
+    o.movie_type,
+    o.visa,
+    o.releaseDate,
```
[the remark this answers is not in the document]
Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql:79]: Yes !

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__mediation.sql [hunk 8]
```diff
@@ -1,3 +1,10 @@
+{% set target_name = target.name %}
+{% set target_schema = generate_schema_name("analytics_" ~ target_name) %}
+
+{{ config(
+    pre_hook="{{create_humanize_id_function()}}"
+) }}
+
 SELECT
     thumbCount,
     idAtProviders,
```
[the remark this answers is not in the document]
Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__mediation.sql:10]: Je fais ça même si ce n'était pas de cette PR que ce modèle a été créé. J'applique la même règle à toutes les colonnes pour que ce soit homogène :)

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql (hunk 4 above)
Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__offer.sql:42]: yes mais comme ce champs était présent dans enriched_offer_data, je pense qu'il faut le garder pour la migration.

**Review by LucileRainteau — DISMISSED** [review]

**Review by LucileRainteau — APPROVED** [review]

**Pull request merged**